framework/test/unit/state_machine/state_machine.spec.ts

```diff
@@ -205,6 +205,49 @@ describe('state_machine', () => {
 			expect(events).toHaveLength(1);
 			expect(events[0].toObject().topics[0]).toEqual(transaction.id);
 		});
+
+		it('should rollback state if afterCommandExecute fails', () => {
+			const events = [
+				{
+					moduleID: utils.intToBuffer(3, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32), utils.getRandomBytes(20)],
+				},
+				{
+					moduleID: utils.intToBuffer(4, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32), utils.getRandomBytes(20)],
+				},
+				{
+					moduleID: utils.intToBuffer(2, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32)],
+				},
+			];
+
+			events.map(e => eventQueue.add(e.moduleID, e.typeID, e.data, e.topics));
```

Comment:
prefer to use `for...of`. `map` should always have left hand side to receive the result

Reply:
Ahh yes, map always returns!